apis/python/src/tiledbsoma/util_iter.py

```diff
@@ -0,0 +1,119 @@
+import abc
+from typing import TypeVar
+
+import numpy as np
+import pyarrow as pa
+import scipy.sparse as sp
+
+# This package's pybind11 code
+import tiledbsoma.libtiledbsoma as clib
+
+from . import somacore  # to be replaced by somacore package, when available
+from .types import NTuple
+
+
+class TableReadIter(somacore.ReadIter[pa.Table]):
+    """Iterator over Arrow Table elements"""
+
+    def __init__(self, sr: clib.SOMAReader):
+        self.sr = sr
+
+    def __next__(self) -> pa.Table:
+        arrow_table = self.sr.read_next()
+        if arrow_table is None:
+            raise StopIteration
+
+        return arrow_table
+
+    def concat(self) -> pa.Table:
+        """Concatenate remainder of iterator, and return as a single Arrow Table"""
+        return pa.concat_tables(self)
+
+    def close(self) -> None:
+        """TODO: Implement when there is a stateful handle that needs to be closed"""
+        pass
+
+
+RT = TypeVar("RT")
```

Comment:
If `RT` is intended to abbreviate `ReadType` or `ReturnType`, IMHO we could comfortably spell out either of those instead of abbreviating

Reply:
Typical python style for generics uses short variables.  See @thetorpedodog code example in [somacore](https://github.com/single-cell-data/SOMA/blob/257db861deca17eca1bb0637951dedec1e025e1c/python-spec/src/somacore/base.py#L50) for example.

Or another example [here](https://github.com/single-cell-data/SOMA/blob/257db861deca17eca1bb0637951dedec1e025e1c/python-spec/src/somacore/data.py#L200)

Reply:
BTW, if any consolation - I am unsure _why_ this is the style.  I had exactly the same reaction as you when I first saw it!